addon/active-model-serializer.js

```diff
@@ -300,6 +300,24 @@ var ActiveModelSerializer = RESTSerializer.extend({
       }, this);
     }
   },
+
+  extractRelationships: function(modelClass, resourceHash) {
+    modelClass.eachRelationship(function (key, relationshipMeta) {
+      var relationshipKey = this.keyForRelationship(key, relationshipMeta.kind, "deserialize");
+      if (resourceHash.hasOwnProperty(relationshipKey)) {
+        var polymorphicTypeKey = this.keyForRelationship(key) + '_type';
```

Comment:
Doesn't seem to be passing all parameters in `this.keyForRelationship(key)`... also, wouldn't this return `relationship_id`?

I think we should check `relationshipMeta.polymorphic` too. I guess it could break in some special cases otherwise (if you DO have an attribute named `relationship_type`).